src/Network.cpp

```diff
@@ -862,9 +862,41 @@ Network::Netresult Network::get_scored_moves(
     if (ensemble == DIRECT) {
         assert(rotation >= 0 && rotation <= 7);
         result = get_scored_moves_internal(planes, rotation);
+    } else if (ensemble == MULTI_AVG && rotation > 1) {
+        // MULTI_AVG rotation is the # of random rotations to provide
+        int num_rotations = rotation;
+
+        assert(num_rotations >= 1 && num_rotations <= 8);
+        std::vector<int> r_list{0, 1, 2, 3, 4, 5, 6, 7};
+        // Need to account for extra 2 in m_squaresize.
+        std::vector<float> policy((BOARD_SIZE + 2) * (BOARD_SIZE + 1));
+
+        if (num_rotations < 8) {
+            std::shuffle(r_list.begin(), r_list.end(), Random::get_Rng());
+        }
+
+        result = get_scored_moves_internal(planes, r_list[0]);
+
+        for (int r = 1; r < num_rotations; ++r) {
+            Netresult tmpresult = get_scored_moves_internal(planes, r_list[r]);
+            result.winrate += tmpresult.winrate;
+            result.policy_pass += tmpresult.policy_pass;
+
+            for (auto res : tmpresult.first) {
+                policy[res.second] += res.first;
```

Comment:
Pretty sure you'll need to do something like below as the original vertex as `second` is stored now an implicit vector index.
```c++
for (auto idx = size_t{0}; idx < BOARD_SQUARES; idx++) {
  policy[idx] += tmpresult.policy[idx];
}
```